### ISSUE TITLE
Problems Classes

### ISSUE THREAD

**Issue by JonathanWenger** [probabilistic-numerics/probnum#279]
Define via classes what problems are solved by probabilistic numerical methods in `probnum/problems/_problems.py`. Examples of this are

```python
class IVProblem:
    def __init__(ode, y_0):
    ....

class QuadProblem:
    def __init__(f, domain, measure):
    ...

class LinearSystem:
   def __init__(A, b):
   ...
```

**Comment by pnkraemer** [comment]
How about a simple named tuple along the lines of 
```
class IVProblem(NamedTuple):

    ode : "ODE"
    y0 : "RandomVariable"

class QuadProblem(NamedTuple):

    f : callable
    domain : callable  # domain(x) is True: domain(other_x) is False
```

Respectively data classes or the like?

**Comment by JonathanWenger** [comment]
You can use a backport of `dataclasses` in the same way we use one for `cached_property` for Python versions < 3.8: https://pypi.org/project/dataclasses/

In `setup.cfg` we have:
```ini
# Dependencies of the project (semicolon/line-separated):
install_requires =
    numpy<=1.19.1; python_version<="3.6"
    numpy; python_version>"3.6"
    scipy>=1.4
    cached_property; python_version<"3.8"
```
and in the import section of `_random_variable.py`:
```python
try:
    # functools.cached_property is only available in Python >=3.8
    from functools import cached_property
except ImportError:
    from cached_property import cached_property

```

**Comment by pnkraemer** [comment]
neat, thanks for the pointer!
